modules/reindex/src/main/java/org/elasticsearch/reindex/BulkByScrollParallelizationHelper.java

```diff
@@ -114,12 +115,14 @@ static <Request extends AbstractBulkByScrollRequest<Request>> void initTaskState
     ) {
         int configuredSlices = request.getSlices();
         if (configuredSlices == AbstractBulkByScrollRequest.AUTO_SLICES) {
-            ClusterSearchShardsRequest shardsRequest = new ClusterSearchShardsRequest();
-            shardsRequest.indices(request.getSearchRequest().indices());
-            client.admin().cluster().searchShards(shardsRequest, listener.safeMap(response -> {
-                setWorkerCount(request, task, countSlicesBasedOnShards(response));
-                return null;
-            }));
+            client.execute(
+                TransportClusterSearchShardsAction.TYPE,
+                new ClusterSearchShardsRequest(request.getTimeout(), request.getSearchRequest().indices()),
```

Comment:
Nit: should this be 30 seconds to be consistent with the existing behaviour? The change here makes it 1 min.